CleverGirl/CleverGirl/Helper/AEHelper.cs

```diff
@@ -212,295 +177,87 @@ public static bool MeleeDamageOutweighsRisk(float attackerMeleeDam, Mech attacke
-                    float weaponChanceToHit = toHitFromPosition3 * (float)weapon4.First.ShotsWhenFired * weapon4.First.DamagePerShot;
-                    if (weaponChanceToHit > unlikelyWeaponChanceToHit)
-                    {
-                        unlikelyWeaponChanceToHit = weaponChanceToHit;
-                        weapon2 = weapon4;
-                    }
+                    Mod.Log.Debug?.Write($" Skipping weapon {wep.UIName} in ranged set due to no valid ammoModes");
                 }
             }
 
-            if (weapon2 != null)
-            {
-                likelyToHitWeapons.Add(weapon2);
-            }
+            return MakeWeaponAmmoModeSets(condensedWeaponAmmoModes);
 
-            return MakeWeaponSets(likelyToHitWeapons);
         }
 
-        // CLONE OF HBS CODE - LIKELY BRITTLE!
-        public static CalledShotAttackOrderInfo MakeOffensivePushOrder(AbstractActor attackingUnit, AttackEvaluator.AttackEvaluation evaluatedAttack, ICombatant target)
+        public static List<List<CondensedWeaponAmmoMode>> MakeWeaponAmmoModeSets(List<CondensedWeapon> weapons)
         {
-            if (!attackingUnit.CanUseOffensivePush() || !ShouldUnitUseInspire(attackingUnit))
-            {
-                return null;
-            }
-            return MakeCalledShotOrder(attackingUnit, evaluatedAttack, target, true);
-        }
-
-        // CLONE OF HBS CODE - LIKELY BRITTLE!
-        public static bool ShouldUnitUseInspire(AbstractActor unit)
-        {
-            float num = AIUtil.CalcMaxInspirationDelta(unit, true);
-            AITeam aiteam = unit.team as AITeam;
-            if (aiteam == null || !unit.CanBeInspired)
-            {
-                return false;
-            }
-            if (num < BehaviorHelper.GetBehaviorVariableValue(unit.BehaviorTree, BehaviorVariableName.Float_MinimumInspirationDamage).FloatVal)
-            {
-                return false;
-            }
-            float num2 = 1f - aiteam.GetInspirationWindow();
-            return num > aiteam.GetInspirationTargetDamage() * num2;
+            List<List<CondensedWeaponAmmoMode>> permutations = new List<List<CondensedWeaponAmmoMode>>();
+            GeneratePermutations(weapons, 0, new List<CondensedWeaponAmmoMode>(), permutations);
+            return permutations;
         }
 
-        // CLONE OF HBS CODE - LIKELY BRITTLE!
-        public static CalledShotAttackOrderInfo MakeCalledShotOrder(AbstractActor attackingUnit, AttackEvaluator.AttackEvaluation evaluatedAttack, ICombatant target, bool isMoraleAttack)
+        private static void GeneratePermutations(List<CondensedWeapon> weapons, int index, List<CondensedWeaponAmmoMode> currentPermutation, List<List<CondensedWeaponAmmoMode>> permutations)
         {
-
-            Mech mech = target as Mech;
-            if (mech == null || !mech.IsVulnerableToCalledShots() || evaluatedAttack.AttackType == AIUtil.AttackType.Melee || evaluatedAttack.AttackType == AIUtil.AttackType.DeathFromAbove)
```

Comment:
Are you reverting to the basic logic for all of this, and it's just not obvious from the diff? Or does this move elsewhere?

Reply:
I was probably just dumb